packages/p2p-media-loader-core/src/types/index.d.ts

```diff
@@ -0,0 +1,45 @@
+export type StreamType = "main" | "secondary";
+
+export type ByteRange = { start: number; end: number };
+
+export type Segment = {
+  readonly localId: string;
+  readonly externalId: number;
+  readonly url: string;
+  readonly byteRange?: ByteRange;
+  readonly startTime: number;
+  readonly endTime: number;
+};
+
+export type Stream = {
+  readonly localId: string;
+  readonly type: StreamType;
+  readonly index: number;
+  readonly segments: Map<string, Segment>;
```

Comment:
`segments` are the internals of Core. Empty `new Map()` should not be passed from outside. That is why ReadonlyStream looks ugly too.

I think about the following:

```typescript
type StreamWithSegments = { stream: TStream, segments: Map<string, Segment> }

private readonly streams = new Map<string, StreamWithSegments>();

  getStreamByUrl(streamUrl: string): ReadonlyStream<StreamWithSegments> | undefined;

  getStream(streamLocalId: string): ReadonlyStream<StreamWithSegments> | undefined;
```